resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -291,14 +303,17 @@ public DnsNameResolverBuilder hostsFileEntriesResolver(HostsFileEntriesResolver
      * @return a {@link DnsNameResolver}
      */
     public DnsNameResolver build() {
+
+        if (resolveCache != null && (minTtl != 0 || maxTtl != Integer.MAX_VALUE || negativeTtl != 0)) {
+            throw new IllegalArgumentException("resolveCache and TTLs are mutually exclusive");
```

Comment:
IMO, `IllegalStateException` is more about a state machine that ends up in an unexpected state, like a component that's supposed to be started but is not. `IllegalArgumentException` is about wrong input, which is the case here. It's your call, of course, but I disagree :)

OK for using wrappers.